examples/mnist_deep_avgpool_mpi.py

```diff
@@ -0,0 +1,276 @@
+            # from rank 0 to all other processes. This is necessary to ensure consistent 
+            # initialization of all workers when training is started with random weights 
+            # or restored from a checkpoint.  
+            hvd.BroadcastGlobalVariablesHook(0), 
+            # Horovod: adjust number of steps based on number of ranks.
+            #tf.train.StopAtStepHook(train_loops // hvd.size()) 
+            tf.train.StopAtStepHook(train_loops) 
+        ]
+        # Enable soft placement and tracing as needed   
+        config = tf.ConfigProto(
+            allow_soft_placement=True,
+            log_device_placement=True if hvd.rank() == 0 else False,
+            inter_op_parallelism_threads=1) 
+
+        with tf.train.MonitoredTrainingSession(hooks=hooks,
+                                               config=config) as sess:
+            step = 0
+            start = time.time() 
+
+            loss_values=[]
+            test_accuracy=[]
+            while not sess.should_stop():
+              batch = mnist.train.next_batch(FLAGS.batch_size)
+              sess.run(train_step, feed_dict={x: batch[0], y_: batch[1]})
+              step +=1
+              if step % 10 == 0:
+                t = time.time()
+                if hvd.rank() == 0:
+                  print('step %d training accuracy %g %g sec to evaluate' % (step,
+                       sess.run(accuracy, feed_dict={x: batch[0],y_: batch[1]}), 
+                       time.time() - t))
+
+              t = time.time()
+              _, summary, loss = sess.run(
+                  [train_step, merged, cross_entropy],
+                  feed_dict={
+                      x: batch[0],
+                      y_: batch[1],
+                      keep_prob: 0.5
+                  })
+              loss_values.append(loss)
+              if hvd.rank() == 0:
+                print('step %d loss %g %g sec for training step' % (step, loss, time.time() - t ))
+              train_writer.add_summary(summary, step)
+
+              if step==(train_loops//hvd.size()-1) and hvd.rank()==0:
+                x_test=mnist.test.images[:num_test_images]
+                y_test=mnist.test.labels[:num_test_images]
+                print('test accuracy: ',  sess.run(accuracy, feed_dict={
+                                           x: x_test,
+                                           y_: y_test}
+                                           )) 
+                test_accuracy.append(accuracy) 
+            print( "Training finished. Running test")
+
+            return loss_values,test_accuracy
+
+def main(_):
+    train_mnist_cnn(FLAGS)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--data_dir',
+        type=str,
+        default='/tmp/tensorflow/mnist/input_data',
+        help='Directory where input data is stored')
+    parser.add_argument(
```

Comment:
This flag is obsolete. Please remove it.

Reply:
removed